### PR TITLE
MAINT: TIFFReader OME-TIFF Multiplex

### PR DESCRIPTION
**Pull request by vqdang**
Prototype for OME TIFF written for multiple images, and a bit of bug fix.
This is mainly for reference and discussion, feel free to delete/close .

### CODE REVIEW

**Comment by vqdang** [comment]
@John-P  Aside from the adhoc implementation for multiplex OME TIFF no z-axis, no time axis. This is the fix location of what I think is a bug.

https://github.com/TissueImageAnalytics/tiatoolbox/blob/794662db6043c9fa230b8e4a01d7b7dd01a06661/tiatoolbox/wsicore/wsireader.py#L3552-L3585

**Comment by rogertrullo** [comment]
@vqdang @John-P this seems related to https://github.com/TissueImageAnalytics/tiatoolbox/issues/452, if so, maybe we can include the fix on resolution reading in this PR

**Comment by shaneahmed** [comment]
@vqdang you created this branch from quite an old commit. Please can you check this merge? https://github.com/TissueImageAnalytics/tiatoolbox/pull/471/commits/cd1dc048ec0e76b6a2e9a37ad1a0423359cf8313

**Comment by shaneahmed** [comment]
> @vqdang @John-P this seems related to #452, if so, maybe we can include the fix on resolution reading in this PR

Thanks @rogertrullo We are looking into #452 but would keep it separate from this PR as this is only related to adding multiplex imaging support.

File: tiatoolbox/wsicore/wsireader.py [hunk 1]
```diff
@@ -3081,6 +3092,13 @@ def _canonical_shape(self, shape):
                 Shape in YXS order.
 
         """
+        # toilet roll multiplex OME TIFF
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:3095]: @vqdang  Please can you provide a reference to this comment?

Comment (John-P) [reply tiatoolbox/wsicore/wsireader.py:3095]: I believe he is referring to the planar configuration of the TIFF file, see https://awaresystems.be/imaging/tiff/tifftags/planarconfiguration.html

File: tests/test_wsireader.py [hunk 2]
```diff
@@ -1738,20 +1738,6 @@ def test_tiffwsireader_invalid_svs_metadata(sample_svs, monkeypatch):
...
-        wsi.tiff.pages[0],
-        "description",
-        wsi.tiff.pages[0].description.replace(
-            '<Objective ID="Objective:0:0" NominalMagnification="20.0"/>', ""
-        ),
-    )
-    with pytest.raises(KeyError, match="No matching Instrument"):
-        _ = wsi._info()
-
-
```
Comment (John-P) [inline tests/test_wsireader.py:1754]: What was the reason for removing this test?

File: tiatoolbox/wsicore/wsireader.py [hunk 3]
```diff
@@ -3007,6 +3011,13 @@ def __getitem__(self, index):
         while len(index) < len(self.axes):
             index = (*index, slice(None))
 
+        # toilet roll multiplex OME TIFF
```
Comment (John-P) [inline tiatoolbox/wsicore/wsireader.py:3014]: Maybe change to "separate planar configuration (AKA toilet roll)" to use the same terminology in spec documents.

**Review by John-P — APPROVED** [review]

**Comment by John-P** [comment]
This probably needs more thorough testing etc but as a quick fix to get things working this looks good.

**Comment by John-P** [comment]
Closing for now while we create a new PR to improve TIFF support in general.

**Pull request closed without merging**